tests/unittests/test_openai_client_multiple.py

```diff
@@ -0,0 +1,76 @@
+import unittest
+from unittest.mock import AsyncMock, MagicMock, patch
```

Comment:
lets Integration test it?
since other backends are also tested like that.
having this as UT will be bit weird.